apps/web/vibes/soul/primitives/dynamic-form/index.tsx

```diff
@@ -118,8 +124,18 @@ export function DynamicForm<F extends Field>({
 
             return <DynamicFormField field={field} formField={formField} key={formField.id} />;
           })}
-          <div className="flex gap-x-3 pt-3">
-            <SubmitButton name={submitName} value={submitValue}>
+          <div className="flex gap-1 pt-3">
+            {onCancel && (
+              <Button
+                aria-label={`${cancelLabel} ${submitLabel}`}
+                onClick={onCancel}
+                size={buttonSize}
+                variant="tertiary"
+              >
+                {cancelLabel}
+              </Button>
+            )}
```

Comment:
It feels weird to add a cancel button to the dynamic form primitive as opposed to adding it to the component that renders this form

Reply:
What's the pattern here then? Pass it in as a prop (`{ cancelButton: ReactNode }` or via `children`?

Reply:
I need to do some more thinking to form an opinion here. As it is right now it feels like we're coupling the needs of the address form with the dynamic form. I think it's completely fine given the time pressures we're feeling right now but later on I would like to rethink how we handle dynamic form fields. Perhaps the component doesn't even attempt to render a full form and just renders the fields, or perhaps the submit button (and cancel button) can be part of the dynamic data. Not sure just yet, so let's just punt it?

Reply:
I see what you mean there. I also see we could use `children` to allow consumers to define the layout of the buttons – the size and width of the buttons should be different between the address and register customer forms. But yeah, let's punt this decision.